src/main/resources/assets/computercraft/lua/rom/programs/shell.lua

```diff
@@ -114,7 +115,22 @@ else
 end
 
 local function run( _sCommand, ... )
-    local sPath = shell.resolveProgram( _sCommand )
+    local checkvar = { ... }
+    table.insert( checkvar, 1, _sCommand )
+    for k, v in ipairs( checkvar ) do
+        if v:find( "$", 1, true ) == 1 then
+            v = v:sub( 2, -1)
+            if type( tShellenv[v] ) == "string" or type( tShellenv[v] ) == "number" or type( tShellenv[v] ) == "boolean" then
+                v = tostring( tShellenv[v] )
+            elseif type( tShellenv[v] ) == "function" then
```

Comment:
Allowing setenv() to be a function seem like a big can of worms. I'd rather you just restrict to the string/number/boolean